### DISCUSSION TITLE
Clarify status of "smart cut"; testing

### DISCUSSION THREAD

**Discussion by mrienstra** [mifi/lossless-cut#1514] [General]
[README.md#features](https://github.com/mifi/lossless-cut/blob/4a9102b/README.md#features) (in reference to "smart cut") points to #126. Some stats re: #126:
1. Started Feb 2019
2. 68 comments, most recent comment dated January 2022
3. Issue description last updated June 2022
4. Issue description "Update" section last updated March 2022

There is also [issues.md#smart-cut-not-working](https://github.com/mifi/lossless-cut/blob/ee29958/issues.md#smart-cut-not-working).

As I was pulling these threads together, I began to feel that I had a decent picture of the current status, but maybe some dates could be added to #126's issue description? I didn't realize at first that it had been kept reasonably up-to-date, until I checked the edit history. If someone makes the same mistake I did: skimming through the 68 comments to ascertain the current status is a bit daunting.

I was planning to suggest closing #126 & opening a new issue, just to "reboot" the conversation and make it a little easier to navigate, but now not sure that would be helpful.

Re: #126's issue description:
> Other codecs may or may not work

I haven't had any luck using "smart cut" with `webm` (from YouTube), might be nice to explicitly mention formats known not to work. I can obtain and try some more `webm` samples to solidify the "not working" status, maybe from https://github.com/mifi/lossless-cut-fixtures, or elsewhere. Here is the output from `ffmpeg -i foo.webm`:

```plain
Input #0, matroska,webm, from 'foo.webm':
  Metadata:
    ENCODER         : Lavf59.27.100
  Duration: 00:01:22.32, start: -0.007000, bitrate: 2748 kb/s
  Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv, bt709), 1920x1080, SAR 1:1 DAR 16:9, 30 fps, 30 tbr, 1k tbn (default)
    Metadata:
      DURATION        : 00:01:22.300000000
  Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
    Metadata:
      DURATION        : 00:01:22.321000000
```

---

## Testing

Hmm, I was thinking about testing around "smart cut", looks likes tests aren't currently run as part of the build process, and  I don't see any (recent-ish) issues or discussions re: status of testing, might be helpful to have a stub for that. I see the [move to vitest 🚀](https://github.com/mifi/lossless-cut/commit/ef406da) commit, which is exciting!

**Comment by mrienstra** [comment]
Did a little more digging, the underlying `ffmpeg` build for `darwin-arm64` is lacking the `vp9` encoder, opened a PR for that: https://github.com/mifi/ffmpeg-build-script/pull/3

**Comment by mifi** [comment]
Would be nice with some tests for smart cut indeed, but I haven't had time for that. And I haven't really found any fix to the problems that people are having with smart cut, so it's hard to make a good test when I don't know how to implement the actual code to test. Smart cut as it's currently implemented is quite simple. It will just re-encode the part of the segment before the keyframe, and then concat it with the rest of the segment, however as evident, this does not always work perfectly.
